src/utils/helperFunctions.ts

```diff
@@ -128,12 +128,83 @@ export const getChartDataFromTimeseriesData = (
                     ? formattedDate
                     : '',
             caseCount: data.cumulativeCases,
+            caseMovingNDaysCount: getNDaysAverage(
+                countryData,
+                idx,
+                7,
+                timeseriesData,
+                0,
+            ),
+            caseMovingNDaysCountCumulative: getNDaysAverage(
+                countryData,
+                idx,
+                7,
+                timeseriesData,
+                1,
+            ),
         };
     });
 
     return chartData;
 };
 
+const getNDaysAverage = (
+    data: TimeseriesCountryDataRow[] | TimeseriesCaseCountsDataRow[],
+    idx: number,
+    nDays: number,
+    timeseriesCountData:
+        | TimeseriesCaseCountsDataRow[]
+        | TimeseriesCountryDataRow[],
+    cumulative = 0,
+): number | null => {
+    //if first 7 days return null
+    if (
+        nDays > idx + 1 &&
+        compareAsc(data[0].date, timeseriesCountData[7].date) === -1
+    )
+        return null;
+
+    const initialValue = 0;
+    let indexOfdateInTimeSeries = -1;
+
+    // check index of date in timeseries
+    for (let i = 0; i < timeseriesCountData.length; i++) {
+        if (compareAsc(data[idx].date, timeseriesCountData[i].date) === 0) {
+            indexOfdateInTimeSeries = i;
+            break;
+        }
+    }
+
+    //add 7 days prior to first date shown in chart
+    const newData = [
```

Comment:
I'm not sure if I understand this one. Could you please explain why do you need to pass `timeseriesCountData` to the function? Correct me if I'm wrong but I think that the easiest way to remove first 7 days (which are empty) from the 7-day-average-chart is to remove first 7 data points from the chart data that are passed to the Chart component. Maybe I'm missing something here 🤔